src/core/Utils.js

```diff
@@ -147,8 +146,46 @@ function runPromiseSequence (functions, ...args) {
 //   return (!f && 'not a function') || (s && s[1] || 'anonymous')
 // }
 
+function isPreviewReady (fileTypeSpecific) {
+  // list of images that browsers can preview
+  if (/^(jpeg|gif|png|svg|bmp)$/.test(fileTypeSpecific)) {
+    return true
+  }
+  return false
+}
+
+function getArrayBuffer (file) {
+  return new Promise(function (resolve, reject) {
+    var reader = new FileReader()
+    reader.addEventListener('load', function (e) {
+      // e.target.result is an ArrayBuffer
+      var arr = new Uint8Array(e.target.result)
+      resolve(arr)
+    })
+    reader.addEventListener('error', function (err) {
+      console.error('FileReader error' + err)
+      reject(err)
+    })
+    const chunk = file.data.slice(0, 4100)
+    reader.readAsArrayBuffer(chunk)
+  })
+}
+
 function getFileType (file) {
-  return file.type ? file.type.split('/') : ['', '']
+  if (file.type) {
+    return Promise.resolve(file.type.split('/'))
+  }
+
+  return getArrayBuffer(file).then((buffer) => {
```

Comment:
I don't know in what situation `FileReader` would error, but it can—perhaps we should fall back to `['', '']` here if the file couldn't be read for some reason?